src/Oracle/Oracle.cs

```diff
@@ -0,0 +1,379 @@
+            if (!CheckOracleAvaiblable(snapshot, out ECPoint[] oracles)) throw new ArgumentException("The oracle service is unavailable");
+            if (!CheckOracleAccount(oracles)) throw new ArgumentException("There is no oracle account in wallet");
+
+            Interlocked.Exchange(ref CancelSource, new CancellationTokenSource())?.Cancel();
+            new Task(() =>
+            {
+                while (CancelSource?.IsCancellationRequested == false)
+                {
+                    using (var snapshot = Blockchain.Singleton.GetSnapshot())
+                    {
+                        IEnumerator<(ulong RequestId, OracleRequest Request)> enumerator = NativeContract.Oracle.GetRequests(snapshot).GetEnumerator();
+                        while (enumerator.MoveNext() && !CancelSource.IsCancellationRequested)
+                            if (!PendingQueue.TryGetValue(enumerator.Current.RequestId, out OracleTask task) || task.Tx is null)
+                                ProcessRequest(snapshot, enumerator.Current.RequestId, enumerator.Current.Request);
+                    }
+                    Thread.Sleep(500);
+                }
+            }).Start();
+        }
+
+        void OnPersist(StoreView snapshot, IReadOnlyList<Blockchain.ApplicationExecuted> applicationExecutedList)
+        {
+            if (!CheckOracleAvaiblable(snapshot, out ECPoint[] oracles) || !CheckOracleAccount(oracles))
+                OnStop();
+        }
+
+        private bool CheckOracleAvaiblable(StoreView snapshot, out ECPoint[] oracles)
+        {
+            oracles = NativeContract.Designate.GetDesignatedByRole(snapshot, Role.Oracle);
+            return oracles.Length > 0;
+        }
+
+        private bool CheckOracleAccount(ECPoint[] oracles)
+        {
+            var oracleScriptHashes = oracles.Select(u => Contract.CreateSignatureRedeemScript(u).ToScriptHash());
+            var accounts = Wallet?.GetAccounts()
+                .Where(u => u.HasKey && !u.Lock && oracleScriptHashes.Contains(u.ScriptHash))
+                .Select(u => (u.Contract, u.GetKey()))
+                .ToArray();
+            return accounts != null && accounts.Length > 0;
+        }
+
+        [ConsoleCommand("stop oracle", Category = "Oracle", Description = "Stop oracle service")]
+        private void OnStop()
+        {
+            Interlocked.Exchange(ref CancelSource, null)?.Cancel();
+        }
+
+        public void SendResponseSignature(ulong requestId, byte[] txSign, KeyPair keyPair)
+        {
+            var message = keyPair.PublicKey.ToArray().Concat(BitConverter.GetBytes(requestId)).Concat(txSign).ToArray();
+            var sign = Crypto.Sign(message, keyPair.PrivateKey, keyPair.PublicKey.EncodePoint(false)[1..]);
+            var param = "\"" + keyPair.PublicKey.ToArray().ToHexString() + "\", " + requestId + ", \"" + txSign.ToHexString() + "\",\"" + sign.ToHexString() + "\"";
```

Comment:
Transaction hash should also be passed to differentiate between broken signature and response mismatch.

And I think we have a problem if the URL being requested is to return random data. IIUC with the current logic each node will construct a response transaction of its own, it couldn't validate other nodes signatures, so it'll wait for `MaxTaskTimeout` and retry. The result is going to be the same (different on every node, validating other nodes signatures fails), so it'll wait for `MaxTaskTimeout`, etc...

Reply:
One way to solve this is not to request again, but rather return an error (may be with new status code).
Signature for this failback transaction can be provided in the same `submit`, though this adds some overhead.

Reply:
> if the URL being requested is to return random data

Currently, we only support static data.

Reply:
I don't think we can make any assumptions over the data being returned, it's an external system, we don't control it, it can have any sort of weird behavior and this behavior should be handled in some reasonable manner by our module. So if the goal of our module is to produce some oracle response transaction for the request given, it must eventually produce that under any circumstances, if it doesn't in some scenarios (like random data returned) --- it's a bug to me.

Reply:
We can try to request some fixed number of attemts and return `Error` after that.